pymaker/auctions.py

```diff
@@ -899,12 +900,18 @@ def parse_event(self, event):
             return Clipper.KickLog(event_data)
         elif signature == "0x05e309fd6ce72f2ab888a20056bb4210df08daed86f21f95053deb19964d86b1":
             event_data = get_event_data(codec, self.take_abi, event)
-            return Clipper.TakeLog(event_data)
+            self._get_sender_for_eventlog(event_data)
+            return Clipper.TakeLog(event_data, self._get_sender_for_eventlog(event_data))
         elif signature == "0x275de7ecdd375b5e8049319f8b350686131c219dd4dc450a08e9cf83b03c865f":
             event_data = get_event_data(codec, self.redo_abi, event)
             return Clipper.RedoLog(event_data)
         else:
             logger.debug(f"Found event signature {signature}")
 
+    def _get_sender_for_eventlog(self, event_data) -> Address:
+        tx_hash = event_data['transactionHash'].hex()
+        receipt = self.web3.eth.getTransactionReceipt(tx_hash)
+        return Address(receipt['from'])
```

Comment:
This could become misleading if the caller of `take` does so through a contract, e.g. `EOA --> CustomContract --> Clipper.take()`. If `take()` were logging `msg.sender` as the bidder, the event would report `address(CustomContract)`. This technique identifies the taker as `EOA`. I'm not sure which is more interesting for your use case, but just be aware it isn't exactly the same as logging `msg.sender` in the `Take` event.

Reply:
(Rather, it's equivalent to if `Take` included `tx.origin` instead.)

Reply:
Indeed; I'd prefer for `Take` to include a `guy` field, but recording the EOA is sufficient for now and better than nothing.

Reply:
And for our considerations if we change this event--would you then prefer that `guy` be `msg.sender`? I notice we aren't logging `who` either, would that be interesting too? (It seems there's no point to `guy == tx.origin` if you can get that in the manner used here.)

Reply:
For consistency, I'd like it to provide the same data `LibNote` gives us in the `usr` field for `tend` and `dent`.  I'd like to remove this hack in LIQ-2.1; pulling the TX receipt for every `take` is quite expensive.